jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/BlockingWriteCallback.java

```diff
@@ -37,7 +37,13 @@ public WriteBlocker acquireWriteBlocker() throws IOException
     {
         return new WriteBlocker(acquire());
     }
-    
+
+    protected long getIdleTimeout()
+    {
+        // Timeout if sending message takes more than 30 sec
+        return 30000;
```

Comment:
what's the expected behavior of the pathological case we encountered (ES-667839) with this patch? Even better if we can manually exercise that expectation.

Reply:
Do we know if there are any current cases where the timeout needs to be larger than 30sec?

Reply:
+1

@xiaochen-db is this issue reproducible? did you test that this patch fixes the issue?

Reply:
I think the `ws.sendString` would throw the timeout error and release the lock on `DriverConnection`.

Reply:
@chadward-db [this dashboard panel](https://grafana.cloud.databricks.com/goto/QjzFQps4z?orgId=1) records the P99 latency of webapp handling ws messages, which including the time consumed by `sendString()`. You see that is well below 30 seconds even in large period of time.

@davidlindb @alexc-db here is reproduction of jetty threads hanging  https://github.com/databricks/universe/pull/301728 and it proves that overriding the timeout could release the blocking call from being stuck. More discussion at: https://databricks.slack.com/archives/C054MMULH7T/p1682707080602089

Reply:
Xiao, can you describe in the PR description how we're avoiding unintended consequences of this change for existing code? Is DriverConnection the only path. If not, what will be the impact if other paths?

Will this only be deployed to Webapp?

Reply:
This will only be deployed to webapp.

There are [several other paths](https://src.dev.databricks.com/search?q=context:universe+path:webapp+-f:Suite+-f:Test+sendString%28&patternType=standard&sm=0&groupBy=repo) in webapp that uses `sendString` and if they hang over 30 seconds, `java.io.IOException: org.eclipse.jetty.util.SharedBlockingCallback$BlockerTimeoutException` will be thrown. I think that's okay since it is deemed a pathological situation

Reply:
The plan to handle `BlockerTimeoutException` from webapp's usage of `sendString()` sgtm - can you either link that PR or fully describe the plan in the PR description?

Reply:
@davidlindb please checkout this wip [pr](https://github.com/databricks/universe/pull/303198)